### ISSUE TITLE
Fireball should log exception content

### ISSUE THREAD

**Issue by ndarilek** [ansible/ansible#2096]
Fireball was crashing for me, and wrote to syslog that an exception had been raised. What it didn't write was the exception itself. It would have helped to diagnose the problem if the exception itself had been logged as well.

**Comment by mpdehaan** [comment]
Can you paste what you saw in syslog?

I have the following code:

```
try:
    daemonize_self(module, password, port, minutes)

    def catcher(signum, _):
        module.exit_json(msg='timer expired')

    signal.signal(signal.SIGALRM, catcher)
    signal.setitimer(signal.ITIMER_REAL, 60 * minutes)


    serve(module, password, port, minutes)
except Exception, e:
    log("exception caught, exiting fireball mode: %s" % e)
    sys.exit(0)
```

I am assuming what got logged was the stringified exception, and I can make it log more, but I am curious what that actual message was just for kicks -- and so I can consider also fixing that at the same time.

Thanks!

**Comment by ndarilek** [comment]
OK, I see the difficulty. I was getting "Operation cannot be completed 
in current state", which didn't resemble an exception in the sense that 
I could follow it through the code and determine what was failing. As it 
stands, that could just as easily be telling me that any given operation 
cannot complete because some generic exception was caught, and said 
generic exception wasn't being revealed to me.

I guess this confusion might only arise for this specific type of 
exception, and if you're unaware that this is an exception rather than a 
generic failure message. I think it would make sense to log the entire 
stacktrace to be entirely unambiguous.

**Comment by mpdehaan** [comment]
Hmm, sounds like it was an internal 0mq exception of some type.

But I agree it should log the full traceback and we should make it do that.

**Comment by mpdehaan** [comment]
fixed in 129e0b8baf264f513264e173b51a3aeddb0f986a
